src/dmd/tocsym.d

```diff
@@ -372,7 +372,8 @@ Symbol *toSymbol(Dsymbol s)
             }
             else
             {
-                final switch (fd.linkage)
+                const l = fd.linkage == LINK.system ? target.systemLinkage() : fd.linkage;
+                final switch (l)
```

Comment:
Could this be deferred to the end of FuncDeclaration's semantic3? Or does that still affect `__traits(getLinkage)`?

Reply:
Yes, for example with an `auto` function:
```D
extern (System) auto foos() {return 0;}
static assert(__traits(getLinkage, foos) == "System");
```